### PR TITLE
Multiplayer testing

### PR DESCRIPTION
**Pull request by naterush**
# Changes

This PR will introduce a technique for testing multiplayer interactions. It does so with a modified design that from what is described in #95.

At a high level: 
- There is a new type of scenario called a "mulitplayer scenario." A mulitplayer scenario is an initial starting state of a notebook, but also a series of "sync steps." When a sync step occurs, "remote commits" get pulled into the local project. These "remote commits" are actually just locally stored objects, and no interactions with the server are necessary.
- To use a mulitplayer scenario in a test, see the example written in the mulitplayer test suite. At a high level, you have to a) start the scenario, and b) tell the scenario when to do the next sync. Otherwise, you write tests like you normally do. 
- The dev screen now allows you to create a simple scenario like before, but also to create a mulitplayer scenario. This process is also documented in the testing README, so @aarondr77 it probably makes sense for you to try to make one (and write a test) to make sure the documentation is consumable.
- To make sure we don't sync randomly and mess up the scenario creation process, the run operation interfaces only turns sync back on when it was on at the start of the operation.

Notably, we do not need to do the messy nonsense described in #95 about saving the step of the scenario in the URL of the project itself. We can just do this in a class, and skip a large number of interactions with the Excel  - on top of things just being cleaner.

- [x] I ran all the tests to make sure I didn't break anything.

### CODE REVIEW

**Comment by naterush** [comment]
The scenario creation and test writing process is currently less ergonomic than I want it to be, but it works for now, IMO.

**Comment by aarondr77** [comment]
@naterush planning on reviewing this tomorrow probably around 5PM or so.

File: src/taskpane/components/DevScreen.js [hunk 1]
```diff
@@ -43,36 +43,163 @@ async function createScenario() {
         fileContents: fileContents
     }));
 }
+ 
+
+export default class DevScreen extends React.Component {
+
+    constructor(props) {
+        super(props);
+        // Don't call this.setState() here!
```
Comment (aarondr77) [inline src/taskpane/components/DevScreen.js]: let's get rid of this comment

File: src/taskpane/components/DevScreen.js [hunk 2]
```diff
@@ -43,36 +43,163 @@ async function createScenario() {
...
+            // We get the incomming data, and append it to the mulitplayer scenario
+            await Excel.run(async (context) => {
+                const project = new Project(context);                
+                const headCommitID = await project.getCommitIDFromBranch(`master`);
+                const parentCommitID = await project.getParentCommitID(headCommitID);
+                const remoteURL = await project.getRemoteURL();
+
+                const update = await getUpdateFromServer(project, remoteURL, headCommitID, parentCommitID);
+                console.log(update);
+                if ("fileContents" in update) {
```
Comment (aarondr77) [inline src/taskpane/components/DevScreen.js]: Can you leave a comment to document what the case where "fileContents" is in update means?

File: src/taskpane/components/DevScreen.js [hunk 3]
```diff
@@ -43,36 +43,163 @@ async function createScenario() {
...
-                <option> Select Secenario</option>
-                {scenarioArray}                
-            </select>
-        </Taskpane>
-    );
-
-
-}
\ No newline at end of file
+const devScreenStyle = {
```
Comment (aarondr77) [inline src/taskpane/components/DevScreen.js:201]: Let's keep css in its own file for continuity.

File: src/tests/README.md [hunk 4]
```diff
@@ -39,6 +41,20 @@ The format of a multi-player testing scenario is a JSON object with the followin
...
 ```
 
-Scenario generation coming soon!
+A `syncStep` represents a "pull from remote" - but as this information is stored in the scenario, we can skip any interact w/ the server and just pull in these commits locally. 
+
+## Creating a Multiplayer Scenario
+
+To create a multiplayer scenario, open (or create) any Saga project and copy its sharing link. Then, use the dev screen to enter a new multiplayer scenario name, and then press "start." This local instance will now record all incoming commits it pulls from remote. 
+
+Then, open a new Excel instance and copy in the sharing link from the existing Saga project. Now, you can make changes in this new Excel instance, and merge them into the shared version. After each merge, or whenever you want to create a new sync step, go back to the original Excel instance, and the console will show it has saved a new sync step.
```
Comment (aarondr77) [inline src/tests/README.md]: Do I need to do something with the sync step in the console?

**Comment by aarondr77** [comment]
I wrote a test and it worked! :) 

It's unfortunate that this framework won't be too helpful for testing long-term documents.

File: src/taskpane/components/DevScreen.js (hunk 3 above)
Comment (naterush) [reply src/taskpane/components/DevScreen.js:201]: Gonna leave this for now, but open an issue about ripping our CSS out into useful files (where is it even stored currently?)

**Comment by naterush** [comment]
Resolved. Merging.

**Pull request merged**